Zend/zend_API.h

```diff
@@ -1577,6 +1580,23 @@ ZEND_API ZEND_COLD void zend_argument_value_error(uint32_t arg_num, const char *
 #define Z_PARAM_OBJECT_OR_NULL(dest) \
 	Z_PARAM_OBJECT_EX(dest, 1, 0)
 
+#define Z_PARAM_OBJ_EX2(dest, check_null, deref, separate) \
```

Comment:
I added a `Z_PARAM_OBJ` which uses the `zend_object` type, rather than `zval`. If it was a bad idea, I can revert it, but I felt that it would help the previous efforts to migrate our APIs from `zval`s to `zend_object`.

Reply:
Maybe add a comment above it, so it's obvious what the difference is?